elements/layercontrol/src/components/layer-config.js

```diff
@@ -1,9 +1,10 @@
 import { LitElement, html } from "lit";
 import { getStartVals } from "../helpers";
-import { dataChangeMethod } from "../methods/layer-config";
+import { dataChangeMethod, applyUpdatedStyles } from "../methods/layer-config";
 import { when } from "lit/directives/when.js";
 import _debounce from "lodash.debounce";
-
+import VectorLayer from "ol/layer/Vector";
```

Comment:
In a short discussion with Ahmed, he mentioned we could potentially ignore the import on build. This would not bundle it in the build. Any project using it would have OL as dependency and thus the import/tree shaking should work when building the corresponding application.
Let's in any case discuss with Silvester

Reply:
The base bundle includes the VectorLayer (see https://eox-a.github.io/EOxElements/?path=/docs/elements-eox-map--docs), so no need for the advanced bundle. In any case, we try to not bundle OL stuff in the other elements, since there could be a mismatch between OL versions if e.g. the eox-layercontrol uses a different OL version than the eox-map.

Reply:
Thanks for pointing this out, unfortunately, the `getLayerType` needs a map which is not available on the `layer-config` or it's parent `layer-tools`. This made me reflect on the concept and I think we can support a broader range of layer types. I added this update, let me know what you think:

```js
const supportStyleConfig = "setStyle" in this.layer || "updateStyleVariables" in this.layer;
```